### DISCUSSION TITLE
What does QUIC_STATUS_INVALID_STATE mean when StreamSend failed?

### DISCUSSION THREAD

**Discussion by panpur** [microsoft/msquic#3137] [Q&A] [answered]
Hello,

I'm doing performance testing for my app by sending 1Gb of data to 50+ peers concurrently using API StreamSend.
I noticed there's a time when StreamSend was failed & returning QUIC_STATUS_INVALID_STATE.
This status only happened once & not able to reproduce again, but I'm curious to know what was causing it.

What are possible scenarios that may result in QUIC_STATUS_INVALID_STATE during StreamSend?

Thanks

**Comment by nibanks** [comment, marked as answer]
That error comes from [here](https://github.com/microsoft/msquic/blob/main/src/core/api.c#L1094). In this code path, if the peer shutdown the connection or stream, you'd get `QUIC_STATUS_ABORTED`, but otherwise (i.e. you shutdown them) you get `QUIC_STATUS_INVALID_STATE`.
